Duke.java

```diff
@@ -0,0 +1,216 @@
+                    int id = Integer.valueOf(response.substring(7));
+                    Task task = taskArrayList.get(id - 1);
+                    taskArrayList.remove(id - 1);
+                    System.out.println("Noted. I've removed this task:");
+                    System.out.println(task.toString().substring(2));
+                    System.out.println("Now you have " + String.valueOf(taskArrayList.size()) + " tasks in the list.");
+                    for (int i = id; i < taskArrayList.size() - 1; i++) {
+                        Task temp = taskArrayList.get(i - 1);
+                        temp = new Task(temp.description, i - 2);
+                    }
+                }
+                else {
+                    //taskArrayList.add(new Task(response, count));
+
+                    CustomException exception = new CustomException();
+                    throw exception;
+                }
+            } catch (CustomException exception) {
+                System.out.println("☹ OOPS!!! The description of a todo cannot be empty.");
+            } catch (StringIndexOutOfBoundsException exception) {
+                System.out.println("☹ OOPS!!! The description lacks details.");
+            }
+        }
+        scanner.close();
+    }
+
+
+
+
+    private void saveTasksToFile() {
+        File file = new File(filePath);
+        File directory = file.getParentFile();
+
+        try {
+            // Create the directory if it doesn't exist
+            if (!directory.exists()) {
+                directory.mkdirs();
+            }
+
+            // Create the file if it doesn't exist
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+
+            try (FileOutputStream fos = new FileOutputStream(file);
+                 ObjectOutputStream oos = new ObjectOutputStream(fos)) {
+                // Write the taskArrayList to the file
+                oos.writeObject(taskArrayList);
+            } catch (IOException e) {
+                System.out.println("Error saving tasks to file!");
+                e.printStackTrace();
+            }
+        } catch (IOException e) {
+            System.out.println("Error creating the file or directory!");
+            e.printStackTrace();
+        }
+    }
+
+    private void loadTasksFromFile() {
+        File file = new File(filePath);
+        if (file.exists()) {
+            try (FileInputStream fis = new FileInputStream(file);
+                 ObjectInputStream ois = new ObjectInputStream(fis)) {
+                // Read the taskArrayList from the file
+                taskArrayList = (ArrayList<Task>) ois.readObject();
+            } catch (IOException | ClassNotFoundException e) {
+                // Handle exceptions, e.g., if the file is corrupted
+                taskArrayList = new ArrayList<>();
+                e.printStackTrace();
+            }
+        } else {
+            taskArrayList = new ArrayList<>();
```

Comment:
similar to the above comment for the try-catch block format and after thinking through, I think maybe you can consider moving taskArrayList = new ArrayList<>() before the if block just to make it cleaner! It's perfectly fine right now though.